### ISSUE TITLE
Poor or missing usage help, inconsistent argument handling

### ISSUE THREAD

**Issue by okdana** [phalcon/phalcon-devtools#855]
Hey there, just trying out Phalcon for the first time. I installed the Phalcon DevTools since some of the tutorials i found on-line use them.

Usually the first thing i do when i install a command-line utility is look at the usage help.

```
proserpina:phalcon-devtools % ./phalcon.phar -h
Phalcon DevTools (3.0.1)

Error: -h is not a recognized command.

proserpina:phalcon-devtools % ./phalcon.phar --help
Phalcon DevTools (3.0.1)

Error: --help is not a recognized command.

proserpina:phalcon-devtools % ./phalcon.phar help
Phalcon DevTools (3.0.1)

Error: help is not a recognized command.
```

OK. None of the usual options work. Let's try running it with no options:

```
proserpina:phalcon-devtools % ./phalcon.phar
Phalcon DevTools (3.0.1)

Available commands:
  commands         (alias of: list, enumerate)
  controller       (alias of: create-controller)
  module           (alias of: create-module)
  model            (alias of: create-model)
  all-models       (alias of: create-all-models)
  project          (alias of: create-project)
  scaffold         (alias of: create-scaffold)
  migration        (alias of: create-migration)
  webtools         (alias of: create-webtools)
```

Well, there's the list of commands, but now what? What do the commands do? Maybe they'll take a help option:

```
proserpina:phalcon-devtools % ./phalcon.phar controller -h
Phalcon DevTools (3.0.1)

Error: This command should be invoked inside a Phalcon project directory.  

proserpina:phalcon-devtools % ./phalcon.phar controller --help
Phalcon DevTools (3.0.1)
                                                                             
Error: This command should be invoked inside a Phalcon project directory.

proserpina:phalcon-devtools % ./phalcon.phar controller help
Phalcon DevTools (3.0.1)
                                                                             
Error: This command should be invoked inside a Phalcon project directory.
```

Maybe not. It says something about projects though, maybe i should see what's up with the `project` command:

```
proserpina:phalcon-devtools % ./phalcon.phar project -h
Phalcon DevTools (3.0.1)
                                
Error: Unknown parameter 'h'  

proserpina:phalcon-devtools % ./phalcon.phar project --help
Phalcon DevTools (3.0.1)

Help:
  Creates a project

Usage:
  project [name] [type] [directory] [enable-webtools]

Arguments:
  help	Shows this help text

Example
  phalcon project store simple

Options:
 --name               Name of the new project
 --enable-webtools    Determines if webtools should be enabled [optional]
 --directory=s        Base path on which project will be created [optional]
 --type=s             Type of the application to be generated (cli, micro, simple, modules)
 --template-path=s    Specify a template path [optional]
 --use-config-ini     Use a ini file as configuration file [optional]
 --trace              Shows the trace of the framework in case of exception [optional]
 --help               Shows this help
```

There's something. Some kind of weird confusion/redundancy between arguments/operands and options, but it seems to work at least. Maybe now i can find out what the `controller` command does.

```
proserpina:phalcon-devtools % cd myproject

proserpina:myproject % ../phalcon.phar controller -h
Phalcon DevTools (3.0.1)

Error: Unknown parameter 'h'

proserpina:myproject % ../phalcon.phar controller --help
Phalcon DevTools (3.0.1)
                                   
Error: Unknown parameter 'help'

proserpina:myproject % ../phalcon.phar controller help
Phalcon DevTools (3.0.1)

Help:
  Creates a controller

Usage:
  controller [name] [directory]

Arguments:
  help	Shows this help text

Options:
 --name=s          Controller name
 --namespace=s     Controller's namespace [option]
 --directory=s     Base path on which project is located [optional]
 --output=s        Directory where the controller should be created [optional]
 --base-class=s    Base class to be inherited by the controller [optional]
 --force           Force to rewrite controller [optional]
```

OK, unlike `project`, `controller` only takes `help` i guess. That's strange, because i went back to look at the documentation on phalconphp.com and it tells me to use `--help` whenever i want to get the usage help for a command.

What about some of the other commands?

```
proserpina:myproject % ../phalcon.phar all-models help
Phalcon DevTools (3.0.1)

Options:
 --config=s       Configuration file [optional]
 --models=s       Models directory [optional]
 --schema=s       Name of the schema. [optional]
 --namespace=s    Model's namespace [optional]
 --extends=s      Models extends [optional]
 --force          Force script to rewrite all the models [optional]
 --get-set        Attributes will be protected and have setters/getters [optional]
 --doc            Helps to improve code completion on IDEs [optional]
 --relations      Possible relations defined according to convention [optional]
 --fk             Define any virtual foreign keys [optional]
 --validations    Define possible domain validation according to conventions
 --directory=s    Base path on which project will be created [optional]
 --mapcolumn      Get some code for map columns [optional]
 --abstract       Abstract Model [optional]
```

I suppose that one will remain a mystery 🙍

**Comment by sergeyklay** [comment]
Hello @okdana

Unfortunately usage the help command is not obvious.
Just use  `./phalcon.phar <command_name> help ` for help:

* `./phalcon.phar commands help`
* `./phalcon.phar controller help`
* `./phalcon.phar module help`
* `./phalcon.phar model help`
* `./phalcon.phar all-models help`
* `./phalcon.phar project help`
* `./phalcon.phar scaffold help`
* `./phalcon.phar migration help`
* `./phalcon.phar webtools help`
 
Usage of `--help` will be fixed in one of future Devtools releases.

**Comment by sergeyklay** [comment]
Fixed in the `3.0.x` branch.
